### PR TITLE
LMS-1916 and LMS-1917: extend emit_task to populate new fields

### PR DESCRIPTION
**Pull request by mlynn-lp**
**Description**
- Extend emit task to populate new fields

**Related Issues**
- #LMS-1916 and LMS-1917

**PR Type**
- Enhancement

### CODE REVIEW

**Review by lzabo — APPROVED** [review]

File: lang/en/logstore_xapi.php [hunk 1]
```diff
@@ -89,6 +89,8 @@
 $string['logstorexapierrorlog'] = 'Logstore xAPI Error Log';
 $string['noerrorsfound'] = 'No errors found';
 $string['logstorexapi:viewerrorlog'] = 'View xAPI error log';
+$string['failed_transform_errortype'] = '101';
```
Comment (gordonmacqueen-lp) [inline lang/en/logstore_xapi.php]: 101?

File: src/loader/moodle_curl_lrs.php [hunk 2]
```diff
@@ -26,6 +26,9 @@
 use src\loader\utils as utils;
 
 function load(array $config, array $events) {
+    unset($_SESSION["xapi_errortype"]);
+    unset($_SESSION["xapi_response"]);
+
```
Comment (gordonmacqueen-lp) [inline src/loader/moodle_curl_lrs.php]: should these lines be removed?

File: lang/en/logstore_xapi.php (hunk 1 above)
Comment (mlynn-lp) [reply lang/en/logstore_xapi.php]: In the absence of an official error code I use 101 as a distinct error code.

File: src/loader/moodle_curl_lrs.php (hunk 2 above)
Comment (mlynn-lp) [reply src/loader/moodle_curl_lrs.php]: Yes, I have removed these.

File: src/loader/utils/load_in_batches.php [hunk 3]
```diff
@@ -29,6 +29,13 @@ function load_in_batches(array $config, array $events, callable $loader) {
 
     // Flags events that weren't transformed successfully as events that didn't load.
     $failedtransformevents = filter_transformed_events($events, false);
+
+    // Add error code.
+    foreach ($failedtransformevents as $failed) {
+        $failed["event"]->errortype = get_string('failed_transform_errortype', 'logstore_xapi');
+        $failed["event"]->response =  get_string('failed_transform_response', 'logstore_xapi');
```
Comment (lzabo) [inline src/loader/utils/load_in_batches.php]: 1. remove extra space before get_string.
2. perhaps default values should be defined in this script and shouldn't be lang string.

**Review by gordonmacqueen-lp — APPROVED** [review]

File: lang/en/logstore_xapi.php (hunk 1 above)
Comment (stephen-oh) [reply lang/en/logstore_xapi.php]: I think we want to remove the underscores here, you can keep underscores below for privacy:metadata:logstore_xapi_log

Comment (mlynn-lp) [reply lang/en/logstore_xapi.php]: I have fixed this.

File: src/loader/utils/load_in_batches.php (hunk 3 above)
Comment (mlynn-lp) [reply src/loader/utils/load_in_batches.php]: I have removed the extra space before get_string.

**Review by stephen-oh — APPROVED** [review]

**Review by gordonmacqueen-lp — APPROVED** [review]

**Review by gordonmacqueen-lp — APPROVED** [review]

**Pull request merged**